File: _exercises/06-word-games/06_word-games.py
# %%
import chatlas
import dotenv
from pyhere import here

dotenv.load_dotenv()

words_file = here() / "_exercises" / "06-word-games" / "words.txt"
words = words_file.read_text().strip().split("\n")

# %%
chat = chatlas.ChatOpenAI(
    system_prompt=f"""
We are playing a word guessing game.
Here's the secret words: {random.sample(words, 1)}.
Give the user an initial riddle and then only answer their questions with yes or no.
When they win, use lots of emojis.
  """
)

# %%
print('Start the chat by saying "Let\'s play"')
chat.console()

File: _exercises/06-word-games/06_word-games.R
library(ellmer)

words <- readLines(
  here::here("_exercises/06-word-games/words.txt")
)

chat <- chat_openai(
  system_prompt = interpolate(
    r"--(
We are playing a word guessing game.
Here's the secret words: {{ sample(words, 1) }}.
Give the user an initial clue and then only answer their questions with yes or no.
When they win, use lots of emojis.
  )--"
  )
)

# Start the chat and say "Let's play"
live_console(chat)
